graviti/portex/factory.py

```diff
@@ -15,18 +16,118 @@
     List,
     Optional,
     Tuple,
+    Type,
     TypeVar,
     Union,
 )
 
 import graviti.portex.ptype as PTYPE
-from graviti.portex.base import PortexType
-from graviti.portex.field import Fields
+from graviti.portex.base import PortexRecordBase, PortexType
+from graviti.portex.builtin import record
+from graviti.portex.field import ConnectedFields, Fields, ImmutableFields, UnionFields
 from graviti.portex.package import Imports
 
 _C = TypeVar("_C", str, float, bool, None)
 
 
+class ConnectedFieldsFactory:
+    """The factory for ConnectedFields.
+
+    Arguments:
+        decl: A dict which indicates a portex type.
+        imports: The :class:`Imports` instance to specify the import scope of the template.
+        type_factory: The :class:`TypeFactory` instance.
+
+    """
+
+    def __init__(
+        self,
+        decl: Dict[str, Any],
+        class_: Type[PortexRecordBase],
+        imports: Imports,
+        type_factory: "TypeFactory",
+    ) -> None:
+        factories: List[Tuple[Optional[str], Callable[..., UnionFields]]] = []
+        fields_factories = (
+            [("fields", lambda **kwargs: kwargs["fields"])]
+            if class_ is record
+            else class_._fields_factory._factories  # type: ignore[has-type]
+        )
+        for key, fields_factory in fields_factories:
+            if key is None:
+                factories.append(
+                    (
+                        None,
+                        partial(
+                            self._get_internal_factory(fields_factory), type_factory=type_factory
```

Comment:
pass the `get_kwargs` method into `_get_internal_factory` rather than `type_factory`?